coreax/metrics.py

```diff
@@ -91,8 +88,8 @@ def compute(
         reference_data: _Data,
         comparison_data: _Data,
         *,
-        block_size: int | None | tuple[int | None, int | None] = None,
-        unroll: tuple[int | bool, int | bool] = (1, 1),
+        block_size: Union[int, None, tuple[Union[int, None], Union[int, None]]] = None,
```

Comment:
I guess the choice not to use `Optional` around a `Union` of the `int` and `tuple` is to avoid excessive nesting?

Reply:
Yep it seemed excessive!